jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +308,37 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+        if display_unit.physical_type != 'length':
+            self.hub.broadcast(SnackbarMessage('Spectral axis unit physical type is '
+                                               f'{display_unit.physical_type}, must be length',
+                                               color="error", sender=self))
+            return
+
+        masks_float_values = np.zeros_like(flux_cube.flux.value, dtype=np.float32)
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+
+        # Loop through cube and create cone aperture at each wavelength. Then convert that to a
+        # mask using the selected aperture method and add that to a mask cube.
+        # TODO: Use flux_cube.spectral_axis.to_value(display_unit) when we have unit conversion.
+        radii = ((flux_cube.spectral_axis.value / self.reference_wavelength) *
+                 self.aperture.selected_spatial_region.radius)
+        im_shape = (flux_cube.shape[1], flux_cube.shape[0])  # Reversed like center
+        aper_method = self.aperture_method_selected.lower()
+        for index, radius in enumerate(radii):
+            aperture = CircularAperture(center, r=radius)
+            slice_mask = aperture.to_mask(method=aper_method).to_image(im_shape)
+            masks_float_values[:, :, index] = ~(slice_mask > 0)
```

Comment:
Oh, I think you turn it into boolean here anyway (even before it hits `NDDataArray`). Is this an oversight?

```python
In [1]: import numpy as np

In [2]: a = np.random.random((2, 2))

In [3]: a
Out[3]:
array([[0.44733494, 0.40654263],
       [0.08520389, 0.97596535]])

In [4]: ~(a > 0)
Out[4]:
array([[False, False],
       [False, False]])
```